include/bout/petsclib.hxx

```diff
@@ -75,7 +77,10 @@ public:
    * The arguments will be passed to PetscInitialize()
    */ 
   static void setArgs(int &c, char** &v) { pargc = &c; pargv = &v;}
-  
+
+  /// Create a KSP linear solver that uses the options specific to this PetscLib
+  KSP createKSPWithOptions(MPI_Comm& comm);
```

Comment:
Can we just pass the `BoutComm` instead? It's nice to avoid doing things with raw MPI where we can.

Reply:
Just a note that the calling site will need `*BoutComm::getInstance()` to pass a reference to the singleton (as it returns a pointer). Also it can't be `const BoutComm&` because `BoutComm::get` is not `const` (and can't be).

Reply:
`BoutComm` only represents the global communicator. Methods like `Mesh::getXcomm` and `Mesh::getYcomm` return `MPI_Comm`, so I think getting rid of `MPI_Comm` here would need to be part of a bigger overhaul.

Reply:
That's a fair point. Another thing for 5.0: remove raw `MPI_*` usage